assets/app/features/upgrade/steps.factory.js

```diff
@@ -95,5 +98,66 @@
                 }
             ];
         }
+
+        function stepByState(state) {
+            return _.find(factory.steps, function (o) { return o.state === state; });
+        }
+
+        function stepByID(id) {
+            return _.find(factory.steps, function (o) { return o.id === id; });
+        }
+
+        function lastStateForRestore(statusData) {
+            var mapping = {
+                'upgrade_prechecks': 'upgrade-landing',
+                'upgrade_prepare': 'upgrade-landing',
+                'admin_backup': 'upgrade.backup',
+                'admin_repo_checks': 'upgrade.administration-repository-checks',
+                'admin_upgrade': 'upgrade.upgrade-administration-server',
+                'database': 'upgrade.database-configuration',
+                'nodes_repo_checks': 'upgrade.nodes-repositories-checks',
+                'nodes_services': 'upgrade.openstack-services',
+                'nodes_db_dump': 'upgrade.openstack-services',
+                'nodes_upgrade': 'upgrade.upgrade-nodes',
+                'finished': 'TODO(skazi): no UI for this step yet',
+            };
+
+            var currentStep = statusData.current_step,
```

Comment:
True. I left it like this as I was considering moving the mapping somewhere else. I will remove the extra `var`.